# AI virtual cell factories for enhanced and genome-wide target prediction
This repo contains a reconstructed verison for DeepVCF, which is model proposed in our paper **"AI virtual cell factories for enhanced and genome-wide target prediction"**.


## Brief Introduction
DeepVCF is AI-driven framework that integrates comprehensive biological knowledge with experimental data to predict engineering targets at a genome-wide scale. By learning system-level relationships between genes and metabolites, DeepVCF extends the scope of traditional metabolic modelling and enables accurate identification of both metabolic and non-metabolic targets.
![](./fig/figure1.jpg)


## Requirements
Build the environment using the following commands in few minutes.
(We have tested that DeepVCF can run on the latest [PyTorch](https://pytorch.org/get-started/locally/).)
```
conda create -n deepvcf python=3.9 -y
conda activate deepvcf
pip3 install torch torchvision --index-url https://download.pytorch.org/whl/cu126
pip install torch-geometric
pip install scikit-learn matplotlib pandas ipykernel 
```
(optional)[Mamba](https://mamba.readthedocs.io/en/latest/) can be used for faster package installation.

## Data & Code
We provide necessary data and code for running DeepVCF in following structure:
```
.
â”œâ”€â”€ code
â”‚   â””â”€â”€ __pycache__
â”œâ”€â”€ data
â”‚   â”œâ”€â”€ KG
â”‚   â”‚   â”œâ”€â”€ ALL
â”‚   â”‚   â”œâ”€â”€ CGL
â”‚   â”‚   â”œâ”€â”€ ECO
â”‚   â”‚   â””â”€â”€ SCE
â”‚   â”œâ”€â”€ me_data
â”‚   â”‚   â”œâ”€â”€ cross_species_transfer
â”‚   â”‚   â”‚   â”œâ”€â”€ cgl
â”‚   â”‚   â”‚   â””â”€â”€ sce
â”‚   â”‚   â”œâ”€â”€ dataset
â”‚   â”‚   â”œâ”€â”€ ffa
â”‚   â”‚   â”œâ”€â”€ metabolic_gene
â”‚   â”‚   â”œâ”€â”€ non_metabolic_gene
â”‚   â”‚   â””â”€â”€ train_data
â”‚   â”‚       â””â”€â”€ embedding_benchmark
â”‚   â”‚           â”œâ”€â”€ amino_acid_hold_out
â”‚   â”‚           â”œâ”€â”€ carbohydrate_hold_out
â”‚   â”‚           â”œâ”€â”€ cofactors_and_vitamins_hold_out
â”‚   â”‚           â”œâ”€â”€ gene_hold_out_1
â”‚   â”‚           â”œâ”€â”€ gene_hold_out_2
â”‚   â”‚           â”œâ”€â”€ lipid_hold_out
â”‚   â”‚           â”œâ”€â”€ metabolite_hold_out
â”‚   â”‚           â”œâ”€â”€ nucleotide_hold_out
â”‚   â”‚           â”œâ”€â”€ random
â”‚   â”‚           â”œâ”€â”€ random_rev
â”‚   â”‚           â””â”€â”€ secondary_metabolites_hold_out
â”‚   â””â”€â”€ other_data
â”œâ”€â”€ fig
â”œâ”€â”€ script
â””â”€â”€ trained_model

```
see our paper for details.


## To train new DeepVCF from scarch, please run the following script
```
# Modify the config if needed.
python script/train_deepvcf.py
```

## Reproduce
For easily reproduce, we reconstruct the code. \
This version largely reproduce our paper results (see script/tutorials.ipynb).
![](./fig/reproduce.png)


## Using DeepVCF for real-world genome-scale target prediction
see script/tutorials.ipynb for more details.

ğŸ”¥ We have successfully applied DeepVCF to the following cases:
```
1.FFAs overproduction â†’ 6 new non-metabolic KO targets with 66.7% success rate (paper).
2.Taurine overproduction â†’ 18 new OE targets with 60.0% success rate (in preparation).
```

ğŸ”” NOTE: 
```
1.We recommend to use DeepVCF_PreFT in real-world applications.
2.DeepVCF/DeepVCF_PreFT might cause confusion in practical applications by simultaneously prioritizing KO and OE of same gene. (For example, rank one in the top 10, and rank the other in the top 50)
```

## To do list
- [ ] Add more species KG.
- [ ] Integrate automated text-mining pipeline.
- [ ] Add active learning partâ†’â€œlab in the Loopâ€.
- [ ] Refine algorithms.


## Coopration
We welcome co-operation on cell factory design alghrithm development and real-world applications. If you have any questions or suggestions, please feel free to contact us.


## Contact
nsk25@mails.tsinghua.edu.cn.
